### DISCUSSION TITLE
What is the process for repository updates?

### DISCUSSION THREAD

**Discussion by RadAd** [microsoft/winget-cli#2525] [General]
When a new version of some software is released what is the process for updating the repository?

They seem to lag a little behind the original website.

Is it a manual or automated process? Do the updates need to be manually approved?

Other package managers seem to use web scraping to update the repository. Does winget do this?

**Comment by mdanish-kh** [comment]
> When a new version of some software is released what is the process for updating the repository?

A manifest for new version is submitted to the [Windows Package Manager Community Repository](https://github.com/microsoft/winget-pkgs). You can learn more about submitting a manifest [here](https://github.com/microsoft/winget-pkgs#submitting-a-package).

> They seem to lag a little behind the original website.

Many packages are not covered by automation and are updated by community contributors. If you find an outdated package, consider opening an [issue](https://github.com/microsoft/winget-pkgs/issues/new/choose) in the community repo or submit a manifest for it once you have gone through the contributing guidelines.

> Is it a manual or automated process? 

At the moment, only packages released on GitHub are highly automated through [vedant's automation](https://github.com/vedantmgoyal2009/vedantmgoyal2009). There is some level of automation available for other packages that:
- Updates the manifest in case of Installer SHA-256 hash mismatch.
- If the publisher uses a static URL for its releases, the manifest gets updated for the new version.

You can see this in action [here](https://github.com/microsoft/winget-pkgs/pulls/wingetbot).

> Do the updates need to be manually approved?

Yes. All manifests require manual review by [community moderators](https://github.com/microsoft/winget-pkgs/blob/master/Moderation.md) before getting published to winget.

> Other package managers seem to use web scraping to update the repository. Does winget do this?

https://github.com/microsoft/winget-pkgs/pull/1467#issuecomment-907429817

**Reply by RadAd** [reply]
I don't understand your reply to my question regarding web scraping. Maybe my question wasn't clear enough.

Other package managers use webscraping to detect when a package is out of date and can automatically update the manifest for it. It doesn't appear that winget does this.

**Reply by mdanish-kh** [reply]
> I don't understand your reply to my question regarding web scraping. Maybe my question wasn't clear enough.
> 
> Other package managers use webscraping to detect when a package is out of date and can automatically update the manifest for it. It doesn't appear that winget does this.

I'm not sure how wingetbot actually works and I didn't find any discussions related to it. But judging from Levvie (moderator)'s comment https://github.com/microsoft/winget-pkgs/pull/1467#issuecomment-907429817, it seems winget does not have the ability to do so. I guess this is related to https://github.com/microsoft/winget-cli/issues/221.

Maybe another mod or @vedantmgoyal2009 can share some insights.

**Reply by vedantmgoyal9** [reply]
The automation updates GitHub packages and also packages whose update information is available by web parsing or via any update APIs.

**Comment by RadAd** [comment]
Thanks for the information, that was very helpful.

It's very disappointing to hear there isn't more automation for this.
